scripts/build-validate.js

```diff
@@ -20,6 +20,16 @@ function validateStandardBuild() {
   assert.ok(fs.existsSync('./dist/index.es.js'));
   assert.ok(fs.existsSync('./dist/styles.css'));
 
+  // Assert it uses named exports rather than properties from the React default
+  // export to help tree-shaking.
+  // React.createElement and React.Fragment are the allowed exceptions
+  const esModuleContent = fs.readFileSync('./dist/index.es.js', 'utf-8');
```

Comment:
The error message is very cryptic when validating against this file (see below), and doesn't contain any actionable feedback.

IMHO the best developer experience would be an ESLint rule with autofix, but I don't know if it's possible.

```
assert.js:85
  throw new AssertionError(obj);
  ^

AssertionError [ERR_ASSERTION]: Input A expected to strictly deep-equal input B:
+ expected - actual

- [
-   'React.PureComponent'
- ]
+ []
    at validateStandardBuild (/Users/kaelig/src/github.com/Shopify/polaris-react/scripts/build-validate.js:31:10)
    at Object.<anonymous> (/Users/kaelig/src/github.com/Shopify/polaris-react/scripts/build-validate.js:11:1)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
child_process.js:656
    throw err;
    ^

Error: Command failed: node scripts/build-validate.js
    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:653:13)
    at run (/Users/kaelig/src/github.com/Shopify/polaris-react/scripts/build.js:5:22)
    at Object.<anonymous> (/Users/kaelig/src/github.com/Shopify/polaris-react/scripts/build.js:27:1)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```